### DISCUSSION TITLE
run tasks without saving file

### DISCUSSION THREAD

**Discussion by RustoMCSpit** [zed-industries/zed#25459] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

i want to run code without saving it

**Comment by notpeter** [comment]
I think we are unlikely to implement this for a number of reasons, but it would be possible with a few scripting languages to pass the script contents via stdin.  Which language(s) were you looking for?

We wouldn't be able to trigger individual functions but we could run the entire script. Also some ambiguity of which directory it should be run from (home dir? project root? `src` dir? etc.).

```
echo 'print("hello world")' | python3 -
echo 'puts "hello world"' | ruby - 
echo 'echo "hello world"' | bash -
echo 'console.log("hello world")' | node -
echo 'print "hello world"' | lua -
echo 'print "hello world\n";' | perl -
echo "Write-Host 'Hello, World!'" | powershell.exe -Command -
```

You could probably do this today with [custom task](https://zed.dev/docs/tasks) and the `ZED_SELECTED_TEXT` variable. Just have a task like `echo ${ZED_SELECTED_TEXT} | python3 -`. Then select all (`cmd-a`) and run.

**Comment by RustoMCSpit** [comment]
Which language(s) were you looking for?


python. if i just want to test a small bit of code, a few lines, i dont want to waste time by saving and then deleting the file. just press f5 and run the code currently visible, it's the sort of thing like in other IDEs like processing, visual studio code, and thonny.

you could probably just create a .temp file and delete it upon closing zed

**Comment by RustoMCSpit** [comment]
mark this as off topic, is there any way to run interpreted language code by pressing a key or clicking on gui (not going into the terminal) like in other IDEs like processing, visual studio code, and thonny. https://github.com/zed-industries/zed/discussions/23307
